sklearn/naive_bayes.py

```diff
@@ -117,6 +119,320 @@ def predict_proba(self, X):
+                                 "but got {}".format(len(model)))
+
+            _, estimator, cols = model
+
+            # Check naive bayes estimator for format
+            # `fit` and `_joint_log_likelihood` attributes
+            if callable(estimator):
+                raise ValueError("Estimator should be a callable specified.")
+            if not (hasattr(estimator, "fit")
+                    or hasattr(estimator, "_joint_log_likelihood")):
+                raise TypeError("Naive bayes estimator should implement "
+                                "the fit and _joint_log_likelihood methods. "
+                                "{} doesn't.".format(type(estimator)))
+            if estimator.__class__.__name__ not in valid_modules:
+                raise ValueError(
+                    "Distributions should be one of {}".format(valid_modules))
+
+            # Check naive bayes estimator for attributes
+            # like `priors`, `class_prior` and `fit_prior`
+            class_prior = getattr(estimator, "priors", None) or \
+                getattr(estimator, "class_prior", None)
+            fit_prior = getattr(estimator, "fit_prior", True)
+            all_class_priors.append(class_prior)
+            all_fit_priors.append(fit_prior)
+
+            # Check the columns for duplicate models and
+            # convert to feature if callable
+            if callable(cols):
+                cols = cols(X)
+            for col in cols:
+                if col in dict_col2model:
+                    raise ValueError("Duplicate specification of "
+                                     f"column {col} found.")
+                else:
+                    dict_col2model[col] = estimator.__class__.__name__.lower()
+            self._cols.append(cols)
+
+        # Check if class priors are the same throughout all estimators
+        if not all(prior is None for prior in all_class_priors):
+            raise ValueError("The parameters 'class_prior' or 'priors' "
+                             "must be the same values throughout all "
+                             "estimators if specified.")
+        if all(prior is not None for prior in all_class_priors):
+            if np.max(np.ptp(all_class_priors, axis=0)) < 1e-6:
+                raise ValueError("The parameters 'class_prior' or 'priors' "
+                                 "must be the same values throughout all "
+                                 "estimators if specified.")
+
+        # Check if `fit_prior`s are the same throughout all estimators
+        if not all(all_fit_priors):
+            raise ValueError("The parameter 'fit_prior' must be "
+                             "the same values through out all estimators "
+                             "if specified.")
+
+        n_features = X.shape[-1]
+        n_cols = len(dict_col2model)
+        if n_cols != n_features:
+            raise ValueError("Expected {} columns".format(n_features) +
```

Comment:
You don't think it should be okay to leave some columns out?

(Also note that ColumnTransformer accepts a `remainder` parameter which would more easily enable the user here to specify one model for categoricals and assume the remainder are numeric... or this might be even more useful in the text classification setting, where you want to use a multinomial or CNB model with lots of features, but accommodate a non-text variable.)

Reply:
I think the `remainder` parameter sounds reasonable. Let me think through this and code up something.